scripts/lib/src/versions.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:pub_semver/pub_semver.dart';
+
+typedef HttpRead = Future<String> Function(Uri url,
+    {Map<String, String>? headers});
```

Comment:
Normally, people just pass around `Client` from `package:http`. Usually, as `http.Client`, if imported with `import 'package:http/http.dart' as http`.

Reply:
Thanks for the tip (and for the offline tip to just not mock this but spin up a test server). I'm not going to do it now, it's probably overkill to make this even more complex.